### PR TITLE
feat: sync route crd labels to apisix

### LINKED ISSUE #775: request help: sync crd labels to apisix
### Issue description
i have some labels like group,project... in k8s crd,  i want to get the labels in apisix plugins. 
### Environment

* your apisix-ingress-controller version (output of `apisix-ingress-controller version --long`);
Version: 1.3.0
Git SHA: d1afc6e
Go Version: go1.17.2
Building OS/Arch: linux/amd64
Running OS/Arch: linux/amd64
* your Kubernetes cluster version (output of `kubectl version`);
* if you run apisix-ingress-controller in Bare-metal environment, also show your OS version (`uname -a`).

### PR DESCRIPTION
**Pull request by shreemaan-abhishek**
### Type of change:

New feature provided

### What this PR does / why we need it:

Closes #775 
### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Review by tao12345666333 — COMMENTED** [review]
Could you please add e2e test case to cover this?

Just create some custom resource, then using admin API to check it's labels.

**Comment by AlinsRan** [comment]
You need to move the following code: `test/e2e/suite-ingress/suite-ingress-features/resourcesync.go` -> `test/e2e/suite-chore/consistency.go`.
The former is to verify the timing synchronization function, and the latter is to verify the consistency of APISIX and controller.

File: test/e2e/suite-ingress/suite-ingress-features/resourcesync.go [hunk 1]
```diff
@@ -320,3 +321,54 @@ spec:
...
+ - name: rule1
+   match:
+     hosts:
+     - httpbin.org
+     paths:
+       - /ip
+   backends:
+   - serviceName: %s
+     servicePort: %d
+   authentication:
```
Comment (AlinsRan) [inline test/e2e/suite-ingress/suite-ingress-features/resourcesync.go:349]: Please remove authentication, it is not used.

File: test/e2e/suite-ingress/suite-ingress-features/resourcesync.go [hunk 2]
```diff
@@ -320,3 +321,54 @@ spec:
...
+			// check if labels exists
+			for _, route := range routes {
+				eq := reflect.DeepEqual(route.Metadata.Labels, labels)
+				assert.True(ginkgo.GinkgoT(), eq)
+			}
+		})
+	}
+
+	ginkgo.Describe("suite-ingress-features: scaffold v2", func() {
+		suites(scaffold.NewScaffold(&scaffold.Options{
```
Comment (AlinsRan) [inline test/e2e/suite-ingress/suite-ingress-features/resourcesync.go:367]: Please refer to it https://github.com/apache/apisix-ingress-controller/blob/master/test/e2e/suite-chore/admin_api.go#L29.
`ApisixResourceSyncInterval` is not required here.

File: test/e2e/suite-ingress/suite-ingress-features/resourcesync.go [hunk 3]
```diff
@@ -320,3 +321,54 @@ spec:
...
 	})
 })
+
+var _ = ginkgo.Describe("suite-ingress-features: apisix labels sync", func() {
+	suites := func(s *scaffold.Scaffold) {
+		ginkgo.JustBeforeEach(func() {
+			labels := map[string]string{"key": "value", "foo": "bar"}
+			backendSvc, backendPorts := s.DefaultHTTPBackend()
+			ar := fmt.Sprintf(`
+apiVersion: apisix.apache.org/v2beta3
```
Comment (AlinsRan) [inline test/e2e/suite-ingress/suite-ingress-features/resourcesync.go:331]: `apiVersion: apisix.apache.org/v2`

**Review by AlinsRan — CHANGES_REQUESTED** [review]

File: test/e2e/suite-chore/consistency.go [hunk 4]
```diff
@@ -273,3 +275,57 @@ var _ = ginkgo.Describe("suite-chore: Consistency between APISIX and the Ingress
...
+			assert.Nil(ginkgo.GinkgoT(), s.EnsureNumApisixRoutesCreated(1), "Checking number of routes")
+
+			routes, _ := s.ListApisixRoutes()
+			assert.Len(ginkgo.GinkgoT(), routes, 1)
+			// check if labels exists
+			for _, route := range routes {
+				eq := reflect.DeepEqual(route.Metadata.Labels, labels)
+				assert.True(ginkgo.GinkgoT(), eq)
+			}
+			// cleanup
```
Comment (shreemaan-abhishek) [inline test/e2e/suite-chore/consistency.go:314]: @AlinsRan is clean up after tests necessary?

File: test/e2e/suite-chore/consistency.go [hunk 5]
```diff
@@ -273,3 +275,57 @@ var _ = ginkgo.Describe("suite-chore: Consistency between APISIX and the Ingress
...
+			for _, route := range routes {
+				_ = s.DeleteApisixRouteByApisixAdmin(id.GenID(route.Name))
+			}
+			// confirm deletion
```
Comment (shreemaan-abhishek) [inline test/e2e/suite-chore/consistency.go:318]: @AlinsRan should I write code for thread sleep to wait for deletion?

File: test/e2e/suite-chore/consistency.go (hunk 4 above)
Comment (shreemaan-abhishek) [reply test/e2e/suite-chore/consistency.go:314]: I added them just in case they are needed, I'll remove them once you review this.

Comment (shreemaan-abhishek) [reply test/e2e/suite-chore/consistency.go]: @AlinsRan, cool! I just removed this part.

**Review by AlinsRan — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]
LGTM

**Pull request merged**